server: fix TestRangeCount

tbg opened the pull request:
Technically fixes a bug but really not one that needs a release note.

The test verifies, or tried to, that the ranges in the system partition
cleanly between "table data" and "rest". There was an unexplained
two-off. The bottom of the rabbit hole reveals:

- tables in the SystemConfigSpan came back as "zero" ranges, so the
  SQL count was missing the SystemConfig range. Ok, add a plus one.
- we burn a tableID on the public schema, which caused another one-off.

It's sort of silly that we split ranges for all of these, but if we
ever change that this test will fail in instructive ways thanks to
all of the debugging I had to put in to reach enlightenment.

cc @celiala not because you should review but because it's a fun blast from the past.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47343)
<!-- Reviewable:end -->

ajwerner reviewed (approved):
`TestAdminAPINonTableStats` seems to need an update

> Reviewed 1 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47343#-:-M4_Nr6JD_V9Q5eJAYbH:b-8ynk1i)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner)

___
*[pkg/server/admin.go, line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47343#-M4_KtQFE02Cd6d9Lm0Z:-M4_KtQGA3EmF6-PU67F:b-8420pd) ([raw file](https://github.com/cockroachdb/cockroach/blob/84a48b485f5f5de3379d7c15e53c647bb03160d6/pkg/server/admin.go#L75)):*
> ```Go
> func nonTableDescriptorRangeCount() int64 {
> 	// NB: explicitly reference them for IDE usage.
> 	var n int64
> ```

what about:

```
return len([]int{
    keys.MetaRangesID, keys.SystemRangesID, keys.TimeseriesRangesID, keys.LivenessRangesID, keys.PublicSchemaID,
})
```


<!-- Sent from Reviewable.io -->

tbg committed:
server: fix MissingTwoRanges

Technically fixes a bug but really not one that needs a release note.

The test verifies, or tried to, that the ranges in the system partition
cleanly between "table data" and "rest". There was an unexplained
two-off. The bottom of the rabbit hole reveals:

- tables in the SystemConfigSpan came back as "zero" ranges, so the
  SQL count was missing the SystemConfig range. Ok, add a plus one.
- we burn a tableID on the public schema, which caused another one-off.

It's sort of silly that we split ranges for all of these, but if we
ever change that this test will fail in instructive ways thanks to
all of the debugging I had to put in to reach enlightenment.

Release note: None

tbg reviewed (commented):
TFTR!

bors r=ajwerner

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47343#-:-M4_eUZs6IqqeP1aG5At:b6i9z8x)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner)



<!-- Sent from Reviewable.io -->

tbg commented:
bors r=ajwerner

Flake on TestCheckConsistencyInconsistent but really looks like system overload, with all heartbeats timing out
